### DISCUSSION TITLE
Is possible to run MS Windows program from DOS program running in DOSEMU?

### DISCUSSION THREAD

**Discussion by fhanzlik** [dosemu2/dosemu2#2199] [General]
I used dosemu (1 and 2) for many years to run my DOS based accounting program. The program is good and still maintained, it is supposed to run under MS Windows, and it has a number of Windows utilities around it, which it runs to perform a number of actions (downloading and uploading files, sending mail, printing, creating PDF outputs and others). Because of these add-ons, I was forced to use the program under MS Windows some time ago. But I would like to return to running under Linux again - if it were possible to run the Windows add-ons as well (some have a Linux equivalent, some I don't need, but some do and they don't have a Linux version).

So the question - is it possible under Linux to run a Windows program from a DOS program running in DOSEMU? 
Without any modification of the DOS program, it is closed.
TIA, Franta Hanzlik

PS: does DOSEMU still have a mailing list? The one that used to be on vger.kernel.org seems to be unavailable (at least some majordomo www interface).

**Comment by stsp** [comment]
There are plenty of ways to do that.
You can run win3.1 under dosemu,
run a dos shell from there, and run
DOS and windows programs under
that DOS shell.
Or you can use HX extender to run
windows progs under DOS directly.
Or you can use unix.com to run your
windows program under wine.

If there is a mail list, no one uses it.
github is much more convenient, and
you can also have a chat in a matrix room:
https://app.element.io/#/room/#dosemu2:matrix.org

**Comment by fhanzlik** [comment]
Hi Stas! Thank you for answer.
But I don't know - the "plenty of ways" doesn't seem like much to me. Those Windows add-ons certainly use many Windows system libraries, use networking and other specifically Windows features - so I think the only way is somehow through Wine (which emulates those Windows libraries). Or am I wrong?

And also - I can't change the DOS accounting program and the way it runs those Windows add-ons.
It would probably be possible to write some fake program-bridge, launched from a DOS program instead of a Windows add-on, which would subsequently launch a Windows add-on under Wine or some Linux equivalent... I don't know...

Does anyone have any experience with this, or recommendations, or can help outright dismiss dead ends?

Thanks, Franta Hanzlik

**Reply by stsp** [reply]
> Or am I wrong?

This can't be guessed, only tried.
Win3.1 networking under dosemu is
possible with trumpet winsock. It is
as well possible under HX. But maybe
something won't work. You just try and
post back.

You can as well run DOS programs
from wine, so maybe wine would be
a full solution for you.

> or can help outright dismiss dead ends?

I don't think so (of course can't speak
for others). Most likely you'll have to
go through a trial and error approach.
